yt_dlp/extractor/yourreformer.py

```diff
@@ -0,0 +1,44 @@
+from .common import InfoExtractor
+
+
+class YourReformerIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?yourreformerathome\.uscreen\.io/programs/(?P<id>[^/?#&]+)'
+    _TESTS = [{
+        'url': 'https://yourreformerathome.uscreen.io/programs/arms-back-abs-7-with-erin?category_id=139586',
+        'info_dict': {
+            'id': 'arms-back-abs-7-with-erin',
+            'ext': 'mp4',
+            'title': 'Arms, Back & Abs 7',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        program_content_url = f'https://yourreformerathome.uscreen.io/programs/{video_id}/program_content'
+
+        program_content = self._download_webpage(
+            program_content_url, video_id,
+            note='Downloading program content')
+        title = self._html_search_regex(
+            r'<h1[^>]*class="[^"]*program-title[^"]*"[^>]*>([^<]+)',
+            program_content, 'title', default=None) or video_id.replace('-', ' ').title()
+        headers = {
+            'referer': 'https://yourreformerathome.uscreen.io/',
+        }
+        m3u8_url = self._search_regex(
+            r'(https?://stream\.mux\.com/[^.]+\.m3u8(?:\?token=[^"\'&]+)?)',
+            program_content, 'm3u8 url')
```

Comment:
Instead of blindly regexing for the m3u8 URL, I would match all `video-players`. If there are multiple they should also be returned as a playlist